frame/dvm/src/lib.rs

```diff
@@ -169,78 +175,7 @@ decl_module! {
 
 			let source = Self::recover_signer(&transaction)
 				.ok_or_else(|| Error::<T>::InvalidSignature)?;
-
-			let transaction_hash = H256::from_slice(
-				Keccak256::digest(&rlp::encode(&transaction)).as_slice()
-			);
-			let transaction_index = Pending::get().len() as u32;
-
-			let (to, contract_address, info) = Self::execute(
-				source,
-				transaction.input.clone(),
-				transaction.value,
-				transaction.gas_limit,
-				Some(transaction.gas_price),
-				Some(transaction.nonce),
-				transaction.action,
-				None,
-			)?;
-
-			let (reason, status, used_gas) = match info {
-				CallOrCreateInfo::Call(info) => {
-					(info.exit_reason, TransactionStatus {
-						transaction_hash,
-						transaction_index,
-						from: source,
-						to,
-						contract_address: None,
-						logs: info.logs.clone(),
-						logs_bloom: {
-							let mut bloom: Bloom = Bloom::default();
-							Self::logs_bloom(
-								info.logs,
-								&mut bloom
-							);
-							bloom
-						},
-					}, info.used_gas)
-				},
-				CallOrCreateInfo::Create(info) => {
-					(info.exit_reason, TransactionStatus {
-						transaction_hash,
-						transaction_index,
-						from: source,
-						to,
-						contract_address: Some(info.value),
-						logs: info.logs.clone(),
-						logs_bloom: {
-							let mut bloom: Bloom = Bloom::default();
-							Self::logs_bloom(
-								info.logs,
-								&mut bloom
-							);
-							bloom
-						},
-					}, info.used_gas)
-				},
-			};
-
-			let receipt = ethereum::Receipt {
-				state_root: match reason {
-					ExitReason::Succeed(_) => H256::from_low_u64_be(1),
-					ExitReason::Error(_) => H256::from_low_u64_le(0),
-					ExitReason::Revert(_) => H256::from_low_u64_le(0),
-					ExitReason::Fatal(_) => H256::from_low_u64_le(0),
-				},
-				used_gas,
-				logs_bloom: status.clone().logs_bloom,
-				logs: status.clone().logs,
-			};
-
-			Pending::append((transaction, status, receipt));
-
-			Self::deposit_event(Event::Executed(source, contract_address.unwrap_or_default(), transaction_hash, reason));
-			Ok(Some(T::GasWeightMapping::gas_to_weight(used_gas.unique_saturated_into())).into())
+			Self::raw_transact(source, transaction)
```

Comment:
The changes here may make synchronizing frontier more complicated in the future, see https://github.com/paritytech/frontier/pull/324. 

Why change to `raw_transact` here?

Reply:
raw_transact will not verify the signature of the transaction.
The system account can't sign the transaction but send it directly.

Reply:
Just take verify signature part out of `do_transact` can also work, right? No need to do more abstract `DvmRawTransactorT` in this approach.

Reply:
yes, can we modify `do_transact `? it may be used by others.

Reply:
I come up with another idea:

1. keep `do_transact` as usual, and add flag to indicate whether to verify signature.

Reply:
you mean we add a flag as a parameter in `do_transact`? I think it's a good idea. it can keep the code structure as consistent as possible with the original.

Reply:
Yes, exactly.